pkg/devfile/generator/utils_test.go

```diff
@@ -774,7 +774,7 @@ func TestGetServiceSpec(t *testing.T) {
 				mockGetComponents.Return(tt.filteredComponents, nil).AnyTimes()
 			}
 			mockDevfileData.EXPECT().GetProjects(common.DevfileOptions{}).Return(nil, nil).AnyTimes()
-
+			mockDevfileData.EXPECT().GetEvents().Return(v1.Events{}).AnyTimes()
```

Comment:
going thru this, I want to confirm with u if `getServiceSpec` should call `GetContainers`. We only want container ports from the main containers right?

Reply:
Yes.  it is calling`GetContainers`, which only returns main containers.